index.js

```diff
@@ -24,9 +28,15 @@ class Chapter {
     return output;
   }
 
-  forEach(array, callback) {
-    for (let i = 0; i < array.length; i++) {
-      callback(array[i]);
+  forEach(sourceItem, callback) {
+    if (typeof sourceItem === 'function') {
+      for (let element of this.result) {
+        sourceItem(element);
+      }
+    } else {
```

Comment:
we should get rid of `else` block here.
this can be done if you `return` inside the previous `if` block.

this will improve the readability a bit